### ISSUE TITLE
Min/max dates are not disabled in popup

### ISSUE THREAD

**Issue by sleptsov** [ng-bootstrap/ng-bootstrap#3598]
### Bug description:

Dates in datepicker popup do not set to disabled state when use dynamic [minDate]/[maxDate] with custom date adapter

In a forked demo from ng-bootstrap Custom date adapter and formatter you can see that min/max is set dynamically from component class and disabled in basic UI mode, but not in a popup  UI mode, if remove (comment out) custom adapter from providers array dates (min/max) are becoming disabled

### Link to minimally-working StackBlitz that reproduces the issue:

https://stackblitz.com/edit/angular-oetchw

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 8.2.14

ng-bootstrap: 5.3.0

Bootstrap: 4.4.1

**Comment by artopaavola** [comment]
This seems to have been working correctly in 5.2.1 and is broken in 5.2.2 ->

**Comment by ar41bald** [comment]
I have the same issue even with the built-in _NgbDateNativeAdapter_. 

### Link to minimally-working StackBlitz that reproduces the issue:
https://stackblitz.com/edit/angular-zdyuru?file=app%2Fdatepicker-popup.ts

As I can see in the date-picker sourcecode the problem is that this Adapter that a developer provides is used to convert \@Input _minDate/maxDate_ (using _toModel_ method) but it probably should not be done in this way as  _minDate/maxDate_ are always _NgbDateStruct_ objects by design.
Here is the line from the _ng-bootstrap_ [source code](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/datepicker/datepicker-input.ts#L443)
```typescript
this._cRef.instance.minDate = this._dateAdapter.toModel(changes.minDate.currentValue);
...
this._cRef.instance.maxDate = this._dateAdapter.toModel(changes.maxDate.currentValue);
```
After these lines executed with the changes like this: `{year: 2020, month: 3, day: 1}`, _minDate/maxDate_ are set to JS _Date_ objects and then they couldn't pass validation in the _[NgbDatePickerService](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/datepicker/datepicker-service.ts#L112)_ and became null
```typescript
    let patch = Object.keys(options)
                    .map(key => this._VALIDATORS[key](options[key]))
                    .reduce((obj, part) => ({...obj, ...part}), {});
```

In my opinion, the solution is - refactor setting _minDate/maxDate_ changes to avoid transforming them by any adapter.

**Comment by maxokorokov** [comment]
@ar41bald thanks for getting to the bottom of this issue. Confirm it was broken in `5.2.2`, apologies it took so long.

Fix should be released in `5.3.1` and `6.0.3`
